### PR TITLE
Fixed leaking http client

### LINKED ISSUE #1587: SAML client creates too many http connections
https://github.com/kolide/fleet/blob/5ad562a952f5e69777b0ae88a7e9ccbe49d15011/server/sso/settings.go#L74-L79

Instead of reusing the client this creates a new one with every request which will cause the server to run out of file descriptors very fast.

### PR DESCRIPTION
**Pull request by murphybytes**
Closes #1587.  The `GetMetadata` method now reuses `http.Client`

### CODE REVIEW

File: server/service/service.go [hunk 1]
```diff
@@ -14,7 +14,9 @@ import (
 	"github.com/kolide/fleet/server/kolide"
 	"github.com/kolide/fleet/server/logwriter"
 	"github.com/kolide/fleet/server/sso"
-	lumberjack "gopkg.in/natefinch/lumberjack.v2"
+	"gopkg.in/natefinch/lumberjack.v2"
+	"net/http"
```
Comment (groob) [inline server/service/service.go]: stdlib imports should go at the top.

File: server/service/service.go [hunk 2]
```diff
@@ -42,6 +44,9 @@ func NewService(ds kolide.Datastore, resultStore kolide.QueryResultStore,
 		osqueryResultLogWriter: resultWriter,
 		mailService:            mailService,
 		ssoSessionStore:        sso,
+		metaDataClient: http.Client{
```
Comment (groob) [inline server/service/service.go]: usually http.Client is a pointer. not sure if it makes a huge difference, but `http.DefaultClient` is also an `&http.Client{}`

File: Gopkg.lock [hunk 3]
```diff
@@ -68,7 +68,7 @@
 
 [[projects]]
   name = "github.com/go-kit/kit"
-  packages = ["endpoint","log","metrics","metrics/internal/lv","metrics/prometheus","transport/grpc","transport/http"]
+  packages = ["endpoint","log","log/level","metrics","metrics/internal/lv","metrics/prometheus","transport/grpc","transport/http"]
```
Comment (groob) [inline Gopkg.lock]: I don't see this new package in the diff. Should Gopkg.lock changes be part of the commit?

**Comment by marpaia** [comment]
Thanks so much for taking this on @murphybytes!

**Pull request merged**